packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -219,126 +208,30 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
     }
 
     constructor(
-        private readonly odspResolvedUrl: IOdspResolvedUrl,
-        private readonly getStorageToken: InstrumentedStorageTokenFetcher,
-        private readonly logger: ITelemetryLogger,
-        private readonly fetchFullSnapshot: boolean,
-        private readonly cache: IOdspCache,
-        private readonly hostPolicy: HostStoragePolicyInternal,
-        private readonly epochTracker: EpochTracker,
-        private readonly flushCallback: () => Promise<FlushResult>,
-        private readonly snapshotFormatFetchType?: SnapshotFormatSupportType,
+        protected readonly odspResolvedUrl: IOdspResolvedUrl,
     ) {
-        this.documentId = this.odspResolvedUrl.hashedDocumentId;
         this.snapshotUrl = this.odspResolvedUrl.endpoints.snapshotStorageUrl;
```

Comment:
Feels like this.snapshotUrl should be handled in same way as all the other variables of that nature (i.e. moved out of base class). Is this mostly about readTree() not checking presence of it directly, but rather calling checkSnapshotUrl() and it to have different implementations for two classes? Or moving check after checking cache? Actually it would be great for call to readTree() to fail (one way or another) if there is no entry in the cache.